apps/somnia/src/commands/archive.ts

```diff
@@ -0,0 +1,263 @@
+import type {
+  ApplicationCommandOptionChoiceData,
+  AutocompleteInteraction,
+  ChatInputCommandInteraction,
+  Interaction,
+  StringSelectMenuInteraction,
+} from 'discord.js'
+import { SlashCommandBuilder } from 'discord.js'
+import * as fs from 'fs'
+import * as path from 'path'
+
+import { error } from '../lib/message'
+import { cwd } from '../util'
+
+import {
+  ArtifactSetKey,
+  LocationCharacterKey,
+  WeaponKey,
+} from '@genshin-optimizer/gi/consts'
+import { artifactarchive } from './archive/artifact'
+import { chararchive } from './archive/char'
+import { weaponarchive } from './archive/weapon'
+
+export const slashcommand = new SlashCommandBuilder()
+  .setName('archive')
+  .setDescription('Genshin Archive')
+  .addSubcommand((s) =>
+    s
+      .setName('char')
+      .setDescription('Characters')
+      .addStringOption((o) =>
+        o
+          .setName('name')
+          .setDescription('Character name')
+          .setAutocomplete(true)
+          .setRequired(true)
+      )
+      .addStringOption((o) =>
+        o.setName('talent').setDescription('Talent name').setAutocomplete(true)
+      )
+  )
+  .addSubcommand((s) =>
+    s
+      .setName('weapon')
+      .setDescription('Weapons')
+      .addStringOption((o) =>
+        o
+          .setName('name')
+          .setDescription('Weapon name')
+          .setAutocomplete(true)
+          .setRequired(true)
+      )
+      .addIntegerOption((o) =>
+        o
+          .setName('refine')
+          .setDescription('Refinement level (1-5)')
+          .setMinValue(1)
+          .setMaxValue(5)
+      )
+  )
+  .addSubcommand((s) =>
+    s
+      .setName('artifact')
+      .setDescription('Artifacts')
+      .addStringOption((o) =>
+        o
+          .setName('name')
+          .setDescription('Artifact set name')
+          .setAutocomplete(true)
+          .setRequired(true)
+      )
+  )
+
+//requiring all the data because imports dont work
+const archivepath = path.join(cwd, '/libs/gi/dm-localization/assets/locales/en')
+//get keys
+const archive: Record<string, any> = {
+  key: {
+    char: require(path.join(archivepath, '/charNames_gen.json')),
```

Comment:
change to import

Reply:
leave a todo to clean this up once translation is properly exported by our libraries